File: exercises/D806/solution.py
def recur(l, r):
    if l == r:
        return
    mid = (l + r) // 2
    recur(l, mid)
    recur(mid + 1, r)
    print(l, mid, mid + 1, r)

n = int(input())
recur(1, n)

File: exercises/D806/solution.cpp
#include <iostream>
using namespace std;
void recur(int l, int r) {
    if (l == r) return ;
    int mid = (l + r) / 2;
    recur(l, mid);
    recur(mid + 1, r);
    cout << l << " " << mid << " " << mid + 1 << " " << r << endl;
}
int main() {
    int n;
    cin >> n;
    recur(1, n);
}
